packages/common/src/api/tan-query/jupiter/useSwapTokens.ts

```diff
@@ -0,0 +1,375 @@
+              outputTokenConfig.mintAddress,
+              actualJupiterDestination
+            )
+          : undefined
+
+        // --- 3e. Handle Output Token (if user bank managed and not directly deposited by Jupiter) ---
+        let wasOutputDepositedToUserBank = false
+        if (
+          outputUserBankAddress &&
+          actualJupiterDestination?.equals(outputUserBankAddress)
+        ) {
+          wasOutputDepositedToUserBank = true
+        }
+
+        const needsManualTransferToOutputUserBank =
+          isOutputUserBankManaged &&
+          actualJupiterDestination &&
+          !wasOutputDepositedToUserBank
+
+        if (needsManualTransferToOutputUserBank) {
+          try {
+            await addAtaToUserBankInstructions({
+              tokenInfo: outputTokenConfig!,
+              userPublicKey,
+              ethAddress: ethAddress!,
+              amountLamports: BigInt(quoteResult.outputAmount.amount),
+              sourceAta: actualJupiterDestination!,
+              sdk,
+              feePayer,
+              instructions
+            })
+            wasOutputDepositedToUserBank = true
+          } catch (error: any) {
+            reportToSentry({
+              name: 'JupiterSwapOutputPostProcessError',
+              error,
+              feature: Feature.TanQuery,
+              additionalInfo: { params }
+            })
+          }
+        }
+
+        // --- 3f. Add Cleanup Instructions for Temporary ATAs ---
+        const atasToClose: PublicKey[] = []
+
+        // Add source ATA if it's temporary
+        if (sourceAtaForJupiter && isSourceAtaTemporary) {
+          atasToClose.push(sourceAtaForJupiter)
+        }
+
+        // Determine if Jupiter setup ATA should be closed
+        const shouldCloseJupiterSetupAta =
+          jupiterTemporarySetupAta &&
+          (!actualJupiterDestination ||
+            !jupiterTemporarySetupAta.equals(actualJupiterDestination)) &&
+          !(
+            needsManualTransferToOutputUserBank &&
+            actualJupiterDestination &&
+            jupiterTemporarySetupAta.equals(actualJupiterDestination)
+          )
+
+        if (shouldCloseJupiterSetupAta) {
+          atasToClose.push(jupiterTemporarySetupAta!)
+        }
+
+        // Add close account instructions for all ATAs that need to be closed
+        for (const ataToClose of atasToClose) {
+          instructions.push(
+            createCloseAccountInstruction(ataToClose, feePayer, userPublicKey)
+          )
+        }
+
+        // ---------- 4. Build and Sign Transaction ----------
+        let swapTx: VersionedTransaction
+        try {
+          swapTx = await sdk.services.solanaClient.buildTransaction({
+            feePayer,
+            instructions,
+            addressLookupTables: addressLookupTableAddresses.map(
+              (addr: string) => new PublicKey(addr)
+            )
+          })
+        } catch (error: any) {
+          reportToSentry({
```

Comment:
I honestly think we can try catch the whole thing and just have one report to sentry at the end